integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/JBossIT.java

```diff
@@ -19,41 +19,56 @@
 package co.elastic.apm.servlet;
 
 import co.elastic.apm.agent.test.AgentTestContainer;
-import co.elastic.apm.servlet.tests.CdiApplicationServerTestApp;
-import co.elastic.apm.servlet.tests.JBossServletApiTestApp;
-import co.elastic.apm.servlet.tests.JavaxExternalPluginTestApp;
-import co.elastic.apm.servlet.tests.JsfApplicationServerTestApp;
-import co.elastic.apm.servlet.tests.SoapTestApp;
-import co.elastic.apm.servlet.tests.TestApp;
+import co.elastic.apm.servlet.tests.*;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
 
 import java.util.Arrays;
 
 @RunWith(Parameterized.class)
 public class JBossIT extends AbstractServletContainerIntegrationTest {
 
-    public JBossIT(final String jbossVersion) {
-        super(AgentTestContainer.appServer("registry.access.redhat.com/" + jbossVersion)
-                .withContainerName("jboss")
-                .withHttpPort(8080)
-                // set JVM arguments through JAVA_OPTS
-                .withJvmArgumentsVariable("JAVA_OPTS")
-                // this overrides the defaults, so we have to manually re-add preferIPv4Stack
-                .withSystemProperty("java.net.preferIPv4Stack", "true")
-                // the other defaults don't seem to be important
-                .withSystemProperty("jboss.modules.system.pkgs", "org.jboss.logmanager,jdk.nashorn.api,com.sun.crypto.provider")
-                .withDeploymentPath("/opt/eap/standalone/deployments"),
+    public JBossIT(final String jbossImage, boolean preserveDefaults) {
+        super(jbossContainer(jbossImage, preserveDefaults),
             "jboss-application");
     }
 
+    private static AgentTestContainer.AppServer jbossContainer(String image, boolean preserveDefaults) {
+        AgentTestContainer.AppServer jboss = AgentTestContainer.appServer(image)
+            .withContainerName("jboss")
+            .withHttpPort(8080)
+            // set JVM arguments through JAVA_OPTS
+            .withJvmArgumentsVariable("JAVA_OPTS")
+
+            .withDeploymentPath("/opt/eap/standalone/deployments")
+            .waitingFor(new HttpWaitStrategy().forPort(8080).forStatusCode(200));
+
+        if (preserveDefaults) {
+            // for older versions setting JAVA_OPTS means overwriting the defaults
```

Comment:
this is only needed for some of the older jboss versions, the latest ones are able to deal with customized JAVA_OPTS variable.